docs/decisions/0027-self-service-pricing.rst

```diff
@@ -0,0 +1,104 @@
+0027 Self-Service Pricing
+*************************
+
+Status
+======
+**In progress** (April 2025)
+
+Context
+=======
+
+For self-service purchasing, multiple frontends/backends/services throughout the customer
+journey will need to know the price of each self-service offering.  It's no longer
+sufficient to store prices in a spreadsheet, as they need to be instantly retrievable.  At
+the same time we can't tolerate price discrepancies within the same checkout journey, so
+prices need to propagate to every system very quickly.
+
+Decision
+========
+
+We will define Stripe products and prices via Terraform.  These prices will be made
+available via a central python API method within enterprise-access which
+dynamically fetches the prices from Stripe, possibly internationalizes it, then
+serializes it in a way that is consumable by frontends.  This python API in
+turn will drive the inclusion of price data within various REST APIs:
+
+* GET https://enterprise-access.edx.org/api/v1/billing/prices
+
+  * Unauthenticated endpoint intended for delivery of pricing information to marketing pages.
+
+* GET https://enterprise-access.edx.org/api/v1/bffs/checkout/checkout
```

Comment:
Looking back at https://github.com/openedx/enterprise-access/pull/680#discussion_r2047979871, I think the endpoint to serve the pricing information (and other logistration field validation rules, etc.) as an unauthenticated user would be:

```
https://enterprise-access.edx.org/api/v1/bffs/checkout/validation/
```

I feel the `checkout` endpoint listed here would be the authenticated `POST` to submit the form / checkout session.